contracts/Comptroller/Diamond/facets/SetterFacet.sol

```diff
@@ -339,11 +372,18 @@ contract SetterFacet is ComptrollerErrorReporter, ExponentialNoError, FacetBase
         return uint(Error.NO_ERROR);
     }
 
+    //----
     function _setTreasuryData(
         address newTreasuryGuardian,
         address newTreasuryAddress,
         uint newTreasuryPercent
-    ) external returns (uint) {
+    )
+        external
+        compareAddress(treasuryGuardian, newTreasuryGuardian)
+        compareAddress(treasuryAddress, newTreasuryAddress)
+        compareValue(treasuryPercent, newTreasuryPercent)
```

Comment:
I don't think this approach is valid in this case. How would we change only the `treasuryPercent`, keeping the other two attributes with the same value?

Reply:
Valid point, my mistake, removing the checks.

Reply:
15311bda0e929dbdde0b67ab13b5d5b3a0b2a81f